src/shared/infra/repositories/schedule_repository_mock.ts

```diff
@@ -697,6 +705,24 @@ export class ScheduleRepositoryMock implements IScheduleRepository {
   async getAllSubjects(): Promise<Subject[]> {
     return this.subjects
   }
+  public getProfessorsByClass(classId: string): User[] {
+
+    const foundClass = this.classes.find(cls => cls.id === classId);
+
+    if (!foundClass) {
```

Comment:
Call getClass() method (all the other methods implement this pattern). 

_NoItemsFound_ is raised on repo for all other methods.

Reply:
We must fix the confusion of NoItemsFoundUsecase and NoItemsFoundRepo later